components/portal/widgets/partials/widget-card.html

```diff
@@ -71,62 +71,8 @@
 
   <!-- BODY -->
   <md-card-content class="widget-content">
-
-    <div ng-switch="widgetType" class="widget-type-container">
-
-      <div ng-switch-when="time-sensitive-content">
-        <time-sensitive app="widget" config="widget.widgetConfig"></time-sensitive>
-      </div>
-
-      <div ng-switch-when="option-link">
-        <option-link app="widget" config="widget.widgetConfig"></option-link>
-      </div>
-
-      <div ng-switch-when="rss">
-        <rss-widget app="widget" config="widget.widgetConfig"></rss-widget>
-      </div>
-
-      <div ng-switch-when="action-items">
-        <action-items app="widget" config="widget.widgetConfig"></action-items>
-      </div>
-
-      <div ng-switch-when="list-of-links">
-        <list-of-links app="widget" config="widget.widgetConfig"></list-of-links>
-      </div>
-
-      <div ng-switch-when="search-with-links">
-        <search-with-links app="widget" config="widget.widgetConfig"></search-with-links>
-      </div>
-
-      <div ng-switch-when="weather">
-        <weather-widget app="widget" config="widget.widgetConfig"></weather-widget>
-      </div>
-
-      <div ng-switch-when="custom" class="custom-widget">
-        <div ng-controller="CustomWidgetController as customWidgetCtrl">
-          <div ng-if="loading" layout="row" layout-align="center center">
-            <loading-gif data-object="content" data-empty="isEmpty"></loading-gif>
-          </div>
-          <content-item ng-if="!loading"></content-item>
-        </div>
-      </div>
-
-      <!-- "basic" widget type-->
-      <div ng-switch-default>
-        <a tabindex="-1" ng-href="{{ renderUrl() }}" target="{{ widget.target ? widget.target : '_self' }}" class="basic-widget" rel="noopener noreferrer">
-          <div class="widget-icon-container" layout="column" layout-align="center center">
-            <widget-icon></widget-icon>
-          </div>
-        </a>
-        <launch-button data-href="{{ renderUrl() }}"
-                       data-target="{{ widget.target ? widget.target : '_self' }}"
-                       data-rel="noopener noreferrer"
-                       data-button-text="{{ widget.widgetConfig.launchText ? widget.widgetConfig.launchText : 'Launch full app' }}"
-                       data-aria-label="{{ 'Launch ' + widget.title }}">
-        </launch-button>
-      </div>
-
-    </div>
+      <widget-content app="widget" config="widget.widgetConfig">
```

Comment:
do you use the config attribute? in the new partial everything gets it from `widget.widgetConfig`

Reply:
I don't use the `config` attribute, at least not intentionally. It's a copy-and-paste emulating something I didn't really understand in the existing calls to the type-specific directives. :) There's maybe some opportunity here to eliminate that, since it's reliably available under `widget` and `widget.widgetConfig` seems to be the preferred way of getting at widget config anyway.